src/radical/pilot/agent/scheduler/base.py

```diff
@@ -217,6 +218,13 @@ def __init__(self, cfg, session):
         rpu.Component.__init__(self, cfg, session)
 
 
+        tmp = os.environ.get('RP_UNIFORM_WORKLOAD', '').lower()
```

Comment:
Why do we care if the workload is uniform or not?

Reply:
Assume you have 1k uniform tasks.  You try to schedule task 1, and it fails.  There is no need to try to schedule task 2 under the same conditions, as it will fail as well.  Or task 3, ...  Knowing that the workload is uniform allows to optimize this, and we only test one task.

In the general case we don't know about uniformity, so this optimization should never apply - but for testing its useful, thus this undocumented env setting to enable this.

Reply:
Ha! I see. I did not think of that specific corner case!